lib/mpl_toolkits/axisartist/grid_finder.py

```diff
@@ -192,25 +192,28 @@ def get_grid_info(self, x1, y1, x2, y2):
 
         grid_info = {
             "extremes": extremes,
-            "lon_lines": lon_lines,
-            "lat_lines": lat_lines,
-            "lon": self._clip_grid_lines_and_find_ticks(
-                lon_lines, lon_values, lon_levs, bb),
-            "lat": self._clip_grid_lines_and_find_ticks(
-                lat_lines, lat_values, lat_levs, bb),
+            # "lon", "lat", filled below.
         }
 
-        tck_labels = grid_info["lon"]["tick_labels"] = {}
-        for direction in ["left", "bottom", "right", "top"]:
-            levs = grid_info["lon"]["tick_levels"][direction]
-            tck_labels[direction] = self._format_ticks(
-                1, direction, lon_factor, levs)
-
-        tck_labels = grid_info["lat"]["tick_labels"] = {}
-        for direction in ["left", "bottom", "right", "top"]:
-            levs = grid_info["lat"]["tick_levels"][direction]
-            tck_labels[direction] = self._format_ticks(
-                2, direction, lat_factor, levs)
+        for idx, lon_or_lat, levs, factor, values, lines in [
+                (1, "lon", lon_levs, lon_factor, lon_values, lon_lines),
+                (2, "lat", lat_levs, lat_factor, lat_values, lat_lines),
+        ]:
+            grid_info[lon_or_lat] = gi = {
+                "lines": [[l] for l in lines],
+                "ticks": {"left": [], "right": [], "bottom": [], "top": []},
+            }
+            for (lx, ly), v, level in zip(lines, values, levs):
+                all_crossings = _find_line_box_crossings(np.column_stack([lx, ly]), bb)
+                for side, crossings in zip(
+                        ["left", "right", "bottom", "top"], all_crossings):
+                    for crossing in crossings:
+                        gi["ticks"][side].append({"level": level, "loc": crossing})
```

Comment:
I wonder if it would make sense for this interior dictionary to be a NumPy record array?

Reply:
Conceptually that makes sense, but I'm not overly convinced by the patch, which is just
```patch
diff --git i/lib/mpl_toolkits/axisartist/grid_finder.py w/lib/mpl_toolkits/axisartist/grid_finder.py
index 897a4152b0..a01a4df53e 100644
--- i/lib/mpl_toolkits/axisartist/grid_finder.py
+++ w/lib/mpl_toolkits/axisartist/grid_finder.py
@@ -208,12 +208,12 @@ class GridFinder:
                 for side, crossings in zip(
                         ["left", "right", "bottom", "top"], all_crossings):
                     for crossing in crossings:
-                        gi["ticks"][side].append({"level": level, "loc": crossing})
+                        gi["ticks"][side].append((level, *crossing, None))
             for side in gi["ticks"]:
-                levs = [tick["level"] for tick in gi["ticks"][side]]
-                labels = self._format_ticks(idx, side, factor, levs)
-                for tick, label in zip(gi["ticks"][side], labels):
-                    tick["label"] = label
+                gi["ticks"][side] = ticks = np.array(
+                    gi["ticks"][side],
+                    [("level", float), ("loc", float, 2), ("angle", float), ("label", object)])
+                ticks["label"] = self._format_ticks(idx, side, factor, ticks["level"])

         return grid_info

diff --git i/lib/mpl_toolkits/axisartist/grid_helper_curvelinear.py w/lib/mpl_toolkits/axisartist/grid_helper_curvelinear.py
index 451e1159ed..d7ed6e8062 100644
--- i/lib/mpl_toolkits/axisartist/grid_helper_curvelinear.py
+++ w/lib/mpl_toolkits/axisartist/grid_helper_curvelinear.py
@@ -83,7 +83,7 @@ class FixedAxisArtistHelper(_FixedAxisArtistHelperBase):
                     (self.nth_coord_ticks, True), (1 - self.nth_coord_ticks, False)]:
                 gi = self.grid_helper._grid_info[["lon", "lat"][nth_coord]]
                 for tick in gi["ticks"][side]:
-                    yield (*tick["loc"], angle_tangent,
+                    yield (*tick[["loc", "angle"]], angle_tangent,
                            (tick["label"] if show_labels else ""))

         return iter_major(), iter([])
@@ -322,8 +322,7 @@ class GridHelperCurveLinear(GridHelperBase):
         lon_or_lat = ["lon", "lat"][nth_coord]
         if not minor:  # major ticks
             for tick in self._grid_info[lon_or_lat]["ticks"][axis_side]:
-                yield *tick["loc"], angle_tangent, tick["label"]
+                yield *tick[["loc", "angle"]], angle_tangent, tick["label"]
         else:
             for tick in self._grid_info[lon_or_lat]["ticks"][axis_side]:
-                yield *tick["loc"], angle_tangent, ""
+                yield *tick[["loc", "angle"]], angle_tangent, ""
```
Thoughts?

Reply:
I think it would be better if we actually used the arrays as a whole, but since the users (in `grid_helper_curvelinear.py`) are just doing a yield on each row, there doesn't seem to be too much benefit here.